### PR TITLE
Nut Cracking

### PR DESCRIPTION
**Pull request by IndiBowstring**
I attached the item placing to ItemFoodFL, we can most likely use it for other recipes (Kneading dough, tenderizing meat, etc)

### CODE REVIEW

File: src/main/java/com/eerussianguy/firmalife/items/ItemFoodFL.java [hunk 1]
```diff
@@ -23,4 +31,18 @@ public ICapabilityProvider getCustomFoodHandler()
     {
         return new FoodHandler(null, data);
     }
+
+    @Override
+    public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
```
Comment (eerussianguy) [inline src/main/java/com/eerussianguy/firmalife/items/ItemFoodFL.java:36]: ??? You can just press shift+V to place an item why would we add this

File: src/main/java/com/eerussianguy/firmalife/items/ItemNutHammer.java [hunk 2]
```diff
@@ -53,6 +59,35 @@ public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos p
                 worldIn.playSound(null, player.getPosition(), SoundEvents.ENTITY_ITEM_BREAK, SoundCategory.PLAYERS, 0.5f, 1.0f);
                 return EnumActionResult.SUCCESS;
             }
+
+            if(worldIn.getBlockState(pos).getBlock() instanceof BlockPlacedItemFlat)
+            {
+                TEPlacedItemFlat tile = (TEPlacedItemFlat) worldIn.getTileEntity(pos);
+                Item item = tile.getStack().getItem();
+
+                if(Constants.RNG.nextInt(2) == 1 && item instanceof IItemFoodTFC) // 50% chance to get a nut
```
Comment (eerussianguy) [inline src/main/java/com/eerussianguy/firmalife/items/ItemNutHammer.java:68]: No reason to check if it's a food item if we're checking the items in the switch statement anyway

File: src/main/java/com/eerussianguy/firmalife/items/ItemNutHammer.java [hunk 3]
```diff
@@ -53,6 +59,35 @@ public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos p
...
+                {
+                    if(tile.getStack().getCapability(CapabilityFood.CAPABILITY, null).isRotten()) {} // Do nothing if the food is rotten
+                    else if(item == ItemsFL.ACORN_FRUIT)
```
Comment (eerussianguy) [inline src/main/java/com/eerussianguy/firmalife/items/ItemNutHammer.java:71]: no registry?

File: src/main/java/com/eerussianguy/firmalife/items/ItemNutHammer.java (hunk 2 above)
Comment (IndiBowstring) [reply src/main/java/com/eerussianguy/firmalife/items/ItemNutHammer.java:68]: Without the instanceof check a nullPointerException is thrown whenever the item interacts with something that isn't an instance of IItemFoodTFC

**Comment by eerussianguy** [comment]
Documenting our talk from discord

- Ought to be a metal tool 
- Check for a quern recipe afterwards
- hammer should spawn items in world rather than giving them out

**Comment by IndiBowstring** [comment]
I've implemented metal mallets.  For now i'm going to keep the quern recipe as an addition to put in later - as its mostly adjacent from the mod's goals.  Pls gib merge

File: src/main/java/com/eerussianguy/firmalife/registry/TFCRegistry.java [hunk 4]
```diff
@@ -12,20 +15,30 @@
...
 import net.dries007.tfc.api.registries.TFCRegistryEvent;
+import net.dries007.tfc.api.types.Metal;
 import net.dries007.tfc.api.types.Ore;
 import net.dries007.tfc.api.types.Plant;
 import net.dries007.tfc.objects.Powder;
 import net.dries007.tfc.objects.inventory.ingredient.IIngredient;
 import net.dries007.tfc.objects.items.ItemPowder;
+import net.dries007.tfc.objects.items.metal.ItemMetal;
+import net.dries007.tfc.api.types.Metal;
+import scala.xml.dtd.ANY;
```
Comment (eerussianguy) [inline src/main/java/com/eerussianguy/firmalife/registry/TFCRegistry.java:38]: ???

**Review by eerussianguy — COMMENTED** [review]
Please delete the stone hammer

File: src/main/java/com/eerussianguy/firmalife/registry/TFCRegistry.java (hunk 4 above)
Comment (IndiBowstring) [reply src/main/java/com/eerussianguy/firmalife/registry/TFCRegistry.java]: Ah, auto import I didn't catch.  Fixed this and removed the stone hammer, as well as updated with current master

**Pull request merged**